Dockerfile

```diff
@@ -13,13 +18,21 @@ WORKDIR /app
 # Copy package.json and package-lock.json
 COPY package.json ./
 COPY pnpm-lock.yaml ./
+COPY scripts ./scripts
+
+ENV GIT_SHA=${GIT_SHA}
+ENV BUILD_TIME=${BUILD_TIME}
 
 # Install dependencies
 RUN pnpm install --frozen-lockfile
 
 # Copy the rest of the application code
 COPY . .
 
+RUN GIT_SHA="${GIT_SHA:-$(git rev-parse --short=12 HEAD 2>/dev/null || echo unknown)}" \
+    BUILD_TIME="${BUILD_TIME:-$(date -u +%Y-%m-%dT%H:%M:%SZ)}" \
```

Comment:
I feel like this is putting too much responsibility into the Dockerfile. 
We could define `GIT_SHA` and `BUILD_TIME` as arg and evaluate them before running `docker build` if needed. But it seems that the `build-version-info.mjs` can also do it by itself, so perhaps we don't need these args at all?